### ISSUE TITLE
Structural equality fail

### ISSUE THREAD

**Issue by CraigStuntz** [fsprojects/FsUnit#78]
This trivial test won't pass:
```fsharp
    [<Test>]
    member this.``structural equality``() =
        let actualList: char list = []
        [(actualList, "")] |> should equal [([], "")]
```
It fails with the interesting error:

>  Expected: [([], "")], 
>    but was [([], "")]

However, it works correctly in FSI:
```fsharp
> let actualList: char list = [];;
   val actualList : char list = []
> [(actualList, "")] = [([], "")];;
   val it : bool = true
```

**Comment by sergey-tihon** [comment]
What unit testing framework do you use? NUnit?

**Comment by CraigStuntz** [comment]
Yes, that's NUnit. FsUnit appears to delegate equality to NUnit, which runs it through [this horror show](https://github.com/nunit/nunit/blob/4e10f475d88fec980f080461a64e6fc4b1e54b2b/src/NUnitFramework/framework/Constraints/NUnitEqualityComparer.cs). It would probably be better if FsUnit did something closer to @forki's [version](https://github.com/forki/FsUnit/blob/master/FsUnit.fs#L10).

**Comment by sergey-tihon** [comment]
Look like similar to this https://github.com/fsprojects/FsUnit/issues/21

**Comment by CraigStuntz** [comment]
It is certainly close to [this comment on that issue](https://github.com/fsprojects/FsUnit/issues/21#issuecomment-57975543), but implementing #21 won't fix the difference in behavior of ```should equal```. I'm thinking that ```x |> should equal y``` and ```x = y``` should be isomorphic.

**Comment by sergey-tihon** [comment]
We should track [discussion on NUnit repo](https://github.com/nunit/nunit/issues/1249)
[FsUnit.Typed](https://github.com/fsprojects/FsUnit/pull/85) should not has this issue.

**Comment by rprouse** [comment]
The NUnit team would appreciate comments and ideas on nunit/nunit#1249. Change will happen if we can come up with a way forward without breaking existing unit tests. As @sergey-tihon says, I don't think this is an FsUnit issue, but that of the underlying framework, NUnit :smile:

**Comment by sergey-tihon** [comment]
Released in [v2.2.0-beta1](https://www.nuget.org/packages/FsUnit/2.2.0-beta1)
